frappe/boot.py

```diff
@@ -221,27 +221,25 @@ def get_allowed_report_names(cache=False) -> set[str]:
 
 
 def get_user_pages_or_reports(parent, cache=False):
+	cache_key = f"has_role:{parent}"
 	if cache:
-		has_role = frappe.cache.get_value("has_role:" + parent, user=frappe.session.user)
```

Comment:
We should keep using get_value here IMO. Separate keys are better for eviction when hitting memory limits (Just today dealing with issue related to this).

Also, hashes are for dictionary-like data _ideally_.